library/spdm_crypt_lib/libspdm_crypt_cert.c

```diff
@@ -1089,31 +1096,35 @@ bool libspdm_x509_common_certificate_check(const uint8_t *cert, size_t cert_size
     /* 4. issuer_name*/
     asn1_buffer_len = 0;
     status = libspdm_x509_get_issuer_name(cert, cert_size, NULL, &asn1_buffer_len);
-    if (asn1_buffer_len == 0) {
+    if ((asn1_buffer_len == 0) && (cert_model != SPDM_CERTIFICATE_INFO_CERT_MODEL_GENERIC_CERT)) {
         status = false;
         goto cleanup;
     }
 
     /* 5. subject_name*/
     asn1_buffer_len = 0;
     status = libspdm_x509_get_subject_name(cert, cert_size, NULL, &asn1_buffer_len);
-    if (asn1_buffer_len == 0) {
+    if ((asn1_buffer_len == 0) && (cert_model != SPDM_CERTIFICATE_INFO_CERT_MODEL_GENERIC_CERT)) {
         status = false;
         goto cleanup;
     }
 
-    /* 6. validaity*/
+    /* 6. validity*/
     status = libspdm_x509_get_validity(cert, cert_size, end_cert_from,
                                        &end_cert_from_len, end_cert_to,
                                        &end_cert_to_len);
     if (!status) {
         goto cleanup;
```

Comment:
if the `Validity` field of cert is **absent** and `cert_model == SPDM_CERTIFICATE_INFO_CERT_MODEL_GENERIC_CERT`,  what is the value of `status` here? true or false?
if the `status` is false,  then even for `GenericCert` certificate model, it will return false, I think it does not make sense.

Reply:
Thanks. I have fixed the code logic.